File: tests/fixtures/header_injection/python/safe_set_header.py
# Safe: request arg routed through `strip_crlf` before being added to the
# response headers.
from flask import request, make_response


def strip_crlf(raw):
    return raw.replace("\r", "").replace("\n", "")


def handler():
    lang = request.args.get("lang")
    safe = strip_crlf(lang)
    resp = make_response("ok")
    resp.headers.add("X-Lang", safe)
    return resp

File: tests/fixtures/header_injection/go/safe_set_header.go
// Safe: query value routed through the project-local `stripCRLF` helper
// before being written to the response header.
package main

import (
	"net/http"
	"strings"
)

func stripCRLF(raw string) string {
	return strings.ReplaceAll(strings.ReplaceAll(raw, "\r", ""), "\n", "")
}

func handler(w http.ResponseWriter, r *http.Request) {
	lang := r.URL.Query().Get("lang")
	safe := stripCRLF(lang)
	w.Header().Set("X-Lang", safe)
}
